components/aws/sagemaker/common/_utils.py

```diff
@@ -407,11 +414,39 @@ def get_image_from_job(client, job_name):
     return image
 
 
+def stop_debug_rules(client, job_name):
+    response = client.describe_training_job(TrainingJobName=job_name)
+    logging.info(response)
+    rules_stopped = []
+    for rule in response["DebugRuleEvaluationStatuses"]:
+        if rule["RuleEvaluationStatus"] == "InProgress" and "RuleEvaluationJobArn" in rule:
+            rule_job_name = convert_rule_arn_to_job_name(rule["RuleEvaluationJobArn"], rule["RuleConfigurationName"])
+            rules_stopped.append(rule_job_name)
+            stop_processing_job(client, rule_job_name)
+    return rules_stopped
+
+
+def convert_rule_arn_to_job_name(rule_evaluation_job_arn, rule_name):
+    """
+
+    Sample inputs and output
+    :param rule_evaluation_job_arn: "arn:aws:sagemaker:us-east-1:123456789000:processing-job/trainingjob-1234567890-rulename-a1112312"
+    :rule_name RuleName
+
+    :output Trainingjob-1234567890-RuleName-a1112312
+    """
+    job_name = rule_evaluation_job_arn.split("/")[1].split("-")
```

Comment:
why not just `rule_evaluation_job_arn.split("/")[1]` ?

Reply:
We want to replace the initial "trainingjob" with "Trainingjob" and change the rule name as well. There is an edge case where the rule name contains the "-" character, just something to note.

Reply:
nit: this method is no longer needed